### ISSUE TITLE
AWS Lambda arm64 Support

### ISSUE THREAD

**Issue by jlarmstrongiv** [lovell/sharp#2938]
#### What are you trying to achieve?

AWS has recently released AWS Graviton2 processor 🎉 up to 19% better performance at 20% lower cost

See also:
  - https://aws.amazon.com/blogs/aws/aws-lambda-functions-powered-by-aws-graviton2-processor-run-your-functions-on-arm-and-get-up-to-34-better-price-performance/
  - https://aws.amazon.com/about-aws/whats-new/2021/09/better-price-performance-aws-lambda-functions-aws-graviton2-processor/
  - https://news.ycombinator.com/item?id=28700209
  - https://www.serverless.com/framework/docs/providers/aws/guide/functions#instruction-set-architecture

Essentially, if I have a docker container starting with:

```dockerfile
FROM public.ecr.aws/lambda/nodejs:14-arm64
```

And then install sharp with `npm init -y && npm install sharp --build-from-source`, we have a crash!

```
#29 33.73 > sharp@0.29.1 install /production-node-modules/node_modules/sharp
#29 33.73 > (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
#29 33.73
#29 36.10 sharp: Installation error: Use with glibc 2.26 requires manual installation of libvips >= 8.11.3
#29 36.11 sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
#29 38.11 npm ERR! code ELIFECYCLE
#29 38.11 npm ERR! errno 1
#29 38.19 npm ERR! sharp@0.29.1 install: `(node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)`
#29 38.19 npm ERR! Exit status 1
#29 38.20 npm ERR!
#29 38.20 npm ERR! Failed at the sharp@0.29.1 install script.
#29 38.20 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
#29 38.26
#29 38.26 npm ERR! A complete log of this run can be found in:
#29 38.26 npm ERR!     /tmp/.npm/_logs/2021-10-20T09_10_33_547Z-debug.log
```

#### Have you searched for similar feature requests?

Yes.  All are closed.

#### What would you expect the API to look like?

Same as regular installation.

#### What alternatives have you considered?

Building from scratch.  Unsolved.  I would like to figure out a workaround to install libvips on AWS Linux 2 ([prior art](https://github.com/libvips/libvips/issues/1184#issuecomment-447973135)).

#### Is there a sample image that helps explain?

N/A

**Comment by jpike88** [comment]
@lovell so after some reading, I understand the 0.30.x will support arm64 on ec2/lambda Amazon Linux 2?

How far are we from a beta or initial build that allows for this support?

**Comment by abhi-precif** [comment]
Here is the only way I have managed to get it working on AWS Graviton 2 EC2 instance with Amazon Linux 2.
(I tried a few other approaches as well, reading all the linked issues regarding this matter). But this is what eventually worked:

Install the necessary dev tools:
```
sudo yum groupinstall "Development Tools"
sudo yum install glib2 glib2-devel
sudo yum install expat expat-devel
```

Install libvips from source:
```
wget https://github.com/libvips/libvips/releases/download/v8.11.3/vips-8.11.3.tar.gz
tar xf vips-8.11.3.tar.gz
cd vips-8.11.3
./configure
make
sudo make install
```

Create a file (let's say `usrlocal.conf`) in `/etc/ld.so.conf.d/`
and add the line `/usr/local/lib` to it .
Run `sudo ldconfig -v`

This would have installed libvips globally which sharp will pick up at the time of installation. Go to your project and run `npm install` as usual.

Not sure if it will help with AWS Lambda but will surely help people struggling with the issue.
